### DISCUSSION TITLE
"Волшебная таблетка" для Windows

### DISCUSSION THREAD

**Discussion by Elllkere** [bol-van/zapret#607] [General]
Решил для всех желающих нажать одну кнопку собрать интерфейс для управления winws с заготовленными пресетами. https://github.com/Elllkere/misterfishdpi

Буду стараться следить за пресетами и обновлять их

**Comment by bol-van** [comment]
Дело неблагодарное за всем этим следить в быстро изменяющейся обстановке.
Но флаг вам а руки

**Comment by iqddd** [comment]
А где список пресетов?

**Reply by Elllkere** [reply]
[Zapret::startProcces()](https://github.com/Elllkere/misterfishdpi/blob/main/ZapretGUI/data.hpp#L146)

**Reply by iqddd** [reply]
list-discord-ip.txt для голосовых соединений Discord'а откуда взят? Сами же понимаете - по IP - это такое себе. Могут поменяться хоть завтра.
tamper там лишнее. Как и диапазон портов 50000-65535. Ставьте 50000-50099.

**Reply by bol-van** [reply]
А без IP можно сломать что-то еще, если оно попадет в те же порты

**Reply by Elllkere** [reply]
> Как и диапазон портов 50000-65535. Ставьте 50000-50099.

Разницы нет, там идет фильтр по IP, но ладно поменяю. Этот пресет в целом не мой, был взят из issues или discussions в день блокировки дискорда, я лишь добавил в winws фильтр по IP (до того как был добавлен ipset usermode)

> list-discord-ip.txt для голосовых соединений Discord'а откуда взят? Сами же понимаете - по IP - это такое себе. Могут поменяться хоть завтра.

Брал [отсюда](https://gist.github.com/AndyIsHereBoi/bf57d7fa1661c82b4a7f5987e56420bf), там же есть и скрипт для получения новых. Как уже сказали выше IP нужны, иначе любое приложение udp с этими портами может отвалиться. Я понимаю, что их могут менять, но это явно будет случаться не часто, а если и сменятся, то можно скриптом получить снова. Можно, конечно, запариться и получать IP в реальном времени, прослушивая wss при подключении, но как-то лень.

**Reply by bol-van** [reply]
Есть еще способ не попортить остальное.
--dpi-desync-fooling=badsum
Он работает и на udp.
Но не факт, что пройдет через ваш домашний роутер. Может и еще где-то застрять.

ttl , возможно, тоже подойдет. Но если будут приходить ttl expired in transit, то сломает некоторые клиенты. Например, chrome ломает, fox - нет. Что с приложениями - по разному.
ttl надо подбирать под провайдера. Autottl невозможен, потому что атака идет на первый пакет, когда входящих еще не было, и длину пути не узнать

**Reply by Elllkere** [reply]
> --dpi-desync-fooling=badsum

В любом случае есть IP фильтр, а тем более уже есть ipset в nfq.

**Reply by iqddd** [reply]
> Есть еще способ не попортить остальное. --dpi-desync-fooling=badsum Он работает и на udp. Но не факт, что пройдет через ваш домашний роутер. Может и еще где-то застрять.
> 
> ttl , возможно, тоже подойдет. Но если будут приходить ttl expired in transit, то сломает некоторые клиенты. Например, chrome ломает, fox - нет. Что с приложениями - по разному. ttl надо подбирать под провайдера. Autottl невозможен, потому что атака идет на первый пакет, когда входящих еще не было, и длину пути не узнать

`--filter-udp=50000-50099 --dpi-desync=fake --dpi-desync-cutoff=d2 --dpi-desync-repeats=6 --dpi-desync-any-protocol`
А у этого есть минусы?

**Reply by bol-van** [reply]
Может поломать другое приложение, если вдруг совпадут порты.
как на это торенты реагируют неизвестно.
Там случайно скачут порты, и какие-то пиры могут неадекватно повести себя на utp

**Reply by iqddd** [reply]
Ну, скажем, вероятность выбора порта для p2p обмена из этого диапазона достаточно мала (там же рандомно выбирается из несколько десятков тысяч). А даже если так, то торрент-клиент, по идее, должен быть готов к нестабильности UDP и, скорее всего, попробует другой порт, переключится TCP или забанит пира, с которым ему не повезло общаться по диапазону 50000-50099.
В теории, не должно сколь-либо сильно повлиять.

**Reply by bol-van** [reply]
Так то так, но все равно потенциальная мина. Может кто-то напороться
